src/linalg/Arnoldi.cpp

```diff
@@ -172,9 +205,80 @@ namespace cytnx {
       }
     }
 
+    void _Arnoldi(std::vector<UniTensor> &out, LinOp *Hop, const UniTensor &UT_init,
+                  const std::string which, const cytnx_uint64 &maxiter, const double &CvgCrit,
+                  const cytnx_uint64 &k, const bool &is_V, const bool &verbose) {
+      int dim = 1;
+      auto UT_init_shape = UT_init.shape();
+      for (int i = 0; i < UT_init.rank(); ++i) {
+        dim *= UT_init_shape[i];
+      }
+      const cytnx_uint64 imp_maxiter = std::min(maxiter, Hop->nx() + 1);
+      const cytnx_complex128 unit_complex = 1.0;
+      auto eigvals = std::vector<Scalar>(k, Scalar());
+      std::vector<UniTensor> eigTens_s;
+      Tensor kry_mat_buffer =
+        cytnx::zeros({imp_maxiter + 1, imp_maxiter + 1}, Hop->dtype(), Hop->device());
+      bool is_cvg = false;
+      auto eigvals_old = std::vector<Scalar>(k, Scalar::maxval(Type.Double));
+      std::vector<UniTensor> buffer;
+      buffer.push_back(UT_init);
+      buffer[0] = buffer[0] / buffer[0].Norm().item();  // normalized q1
+
+      // start arnoldi iteration
+      for (auto i = 1; i < imp_maxiter; i++) {
+        cytnx_uint64 krydim = i;
+        auto nextTens = Hop->matvec(buffer[i - 1]).astype(Hop->dtype());
+        buffer.push_back(nextTens);
+        for (cytnx_uint32 j = 0; j < krydim; j++) {
+          auto h = Dot(buffer[i], buffer[j]).conj();
+          kry_mat_buffer[{i - 1, j}] = h;
+          buffer[i] -= h * buffer[j];
+        }
+        auto h = std::sqrt(static_cast<double>(Dot(buffer[i], buffer[i]).real()));
+        kry_mat_buffer[{i - 1, i}] = h;
+        buffer[i] /= h;
+        Tensor kry_mat = ResizeMat(kry_mat_buffer, krydim, krydim);
```

Comment:
Are you sure we don't have this ResizeMat functionality implemented somewhere?

Reply:
I tried to find it but I could not find. Maybe we could provide this function in the future.

Reply:
create another issue for this

Reply:
OK, see issue #415 .